models/article-model.js

```diff
@@ -1,16 +1,40 @@
-const db = require('../db/connection')
+const db = require("../db/connection");
+
+exports.fetchArticles = () => {
+  return db
+    .query(
+      `
+      SELECT
+          articles.author,
+          title,
+          articles.article_id,
+          topic,
+          articles.created_at,
+          articles.votes,
+          article_img_url,
+          COUNT(comments.article_id) AS comment_count
+      FROM articles
+      LEFT JOIN comments ON articles.article_id = comments.article_id
+      GROUP BY articles.article_id 
+      ORDER BY articles.created_at DESC;
+  `
+    )
+    .then((result) => {
+      return result.rows;
+    });
+};
 
 exports.selectArticle = (article_id) => {
-    return db
+  return db
     .query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
     .then((result) => {
-        const article = result.rows[0];
-        if (!article) {
-          return Promise.reject({
-            status: 404,
-            msg: "No article found!",
-          });
-        }
-        return article;
-      });
-  };
\ No newline at end of file
+      const article = result.rows[0];
+      if (!article || Object.keys.length === 0) {
```

Comment:
Object.keys.length by itself doesn't mean anything, I think you meant - Object.keys(article)
Also, you don't actually need to check the length of the keys since !article is effectively already checking for the same thing ( made a mistake in my last comment, oops 😅  )

Happy to see this merged solo after taking the object.keys bit out!